### DISCUSSION TITLE
Redis session age problem

### DISCUSSION THREAD

**Discussion by hariprasath-yadav** [adonisjs/core#2022] [Help] [answered]
i have setting `age: '2h',`  in session.ts
but in Redis TTL for the key is 7200000 seconds
the milliseconds of the session is converted into seconds in Redis TTL
how do I solve the problem

**Comment by hariprasath-yadav** [comment, marked as answer]
Found 2 Solutions
### Solution 1
just use seconds in interger like `age: 3600,`
## (or)

### Solution 2
change code from
```js
ms_1.default(processedConfig.age
```
to
```js
Math.round(ms_1.default(this.config.age) / 1000)
```
at `<Project Folder>/node_modules/@adonisjs/session/build/src/Drivers/Redis.js` in 25th line.

**Comment by thetutlage** [comment]
Its already been fixed in the latest release https://preview.adonisjs.com/blog/december-2020-release#bug-fixes

**Reply by hariprasath-yadav** [reply]
Yaa right thank you,
but I am using
```js
    "@adonisjs/redis": "^4.1.2",
    "@adonisjs/session": "^3.0.4",
```
packages I cannot upgrade any package to the latest version if I update the package it will show **"Singleton error"** in development from the server what I have to do, should I have to update the @adonisjs/core.

**Reply by thetutlage** [reply]
Looks like you are few releases behind. If you decide to upgrade, begin by following the October release first and then the December one

**Comment by hariprasath-yadav** [comment]
where I can find upgrade docs to upgrade October and December Releases?
Actually, I am in July release

**Reply by thetutlage** [reply]
In the blog https://preview.adonisjs.com/blog

**Reply by hariprasath-yadav** [reply]
thank you very much for your work, support, and kindness.
